### PR TITLE
Ghost Actor Default Timeouts

### PR DESCRIPTION
**Pull request by neonphog**
## PR summary

This creates a default timeout that can be used for most normal use-cases.

Adds builder pattern for GhostTracker, and ContextEndpoints + adds default timeout to both which defaults to 2 seconds.

Separates:

- `bookmark_timeout` from `bookmark` (which uses the default timeout)
- `request_timeout` from `request` (which uses the default timeout)

## Review checklist 
- [ ] The story has unit or integration tests
- [ ] No new bugs, and any tech-debt is identified and justified
- [ ] There is enough API documentation (how to use)
- [ ] There is enough code documentation (how the code works)

### CODE REVIEW

File: crates/ghost_actor/src/ghost_actor.rs [hunk 1]
```diff
@@ -270,12 +273,21 @@ impl<
...
-        timeout: std::time::Duration,
         context: Context,
         payload: RequestToChild,
         cb: GhostCallback<Context, RequestToChildResponse, Error>,
     ) {
-        self.endpoint.request(timeout, context, payload, cb)
+        self.endpoint.request(context, payload, cb)
+    }
+
+    pub fn request_timeout(
```
Comment (struktured) [inline crates/ghost_actor/src/ghost_actor.rs]: Just an idea, not blocker- we could future proof this by making an optional argument container, much like holochain does for various functions. For instance here's the simple `get_entry` invoking the more advanced version:

```Rust
pub fn get_entry(address: &Address) -> ZomeApiResult<Option<Entry>> {
    let entry_result = get_entry_result(address, GetEntryOptions::default())?;
...
}
```

File: crates/ghost_actor/src/ghost_tracker.rs [hunk 2]
```diff
@@ -28,22 +28,50 @@ struct GhostTrackerEntry<Context, CbData, E> {
...
 
-impl<Context, CbData, E> GhostTracker<Context, CbData, E> {
-    /// create a new tracker instance (with request_id prefix)
-    pub fn new(request_id_prefix: &str) -> Self {
+impl Default for GhostTrackerBuilder {
+    fn default() -> Self {
         Self {
-            request_id_prefix: request_id_prefix.to_string(),
+            request_id_prefix: "".to_string(),
+            default_timeout: std::time::Duration::from_millis(2000),
```
Comment (struktured) [inline crates/ghost_actor/src/ghost_tracker.rs]: maybe make a constant for this timeout value, just to make it slightly easier to dig up when debugging timeout conditions. Although perhaps the `Default` trait encapsulates it enough already.

File: crates/ghost_actor/src/ghost_actor.rs (hunk 1 above)
Comment (neonphog) [reply crates/ghost_actor/src/ghost_actor.rs]: :+1: I like that. Updated.

File: crates/ghost_actor/src/ghost_tracker.rs (hunk 2 above)
Comment (neonphog) [reply crates/ghost_actor/src/ghost_tracker.rs]: :+1:

**Pull request merged**